### ISSUE TITLE
Zulu install dependency on Ubuntu repo

### ISSUE THREAD

**Issue by mstormi** [openhab/openhabian#457]
I noticed 'on the go' that Zulu installation routine retrieves keys from an Ubuntu repo.
Now as unavailability of DNS, inet access or repos or changed keys may (and for some people do) result in installation failures, I wonder whether this is really needed or if it can be removed.
I don't know why it is in there (I'm guessing it's a leftover of earlier days when the packages were not yet directly downloaded from Azul Systems) and I'm unsure if there's another dependency I don't see.

**Comment by BClark09** [comment]
The repository packages are signed, apt requires a key to validate the package.

Alternatively, the key can be obtained from the Azul site rather than a key repository with a specific hash.

**Comment by mstormi** [comment]
Ok. Now given we're a little behind on Java (see also #453) and the fact that the Azul repo does not seem to get updated, shouldn't we move to a different installation method (wget download from https://www.azul.com/downloads/zulu-embedded/ instead of apt) ?
AFAIK those packages aren't signed, so we could remove validation, too.

**Comment by BClark09** [comment]
Absolutely, looking at the documentation on Zulu, it seems that zulu-embedded dropped its support for using a repository some time ago.

Ideally, we'd need to change this so that it downloaded, extracted and set as default using `update-alternatives` or `update-java-alternatives`. The harder part, is pointing towards Azul's terms and conditions and grabbing the latest version of Java 8 programmatically.

**Comment by mstormi** [comment]
Agree.
Right on the download page, Azul just says "By downloading software from this page, you agree to the Terms of Use." with "Terms of Use" being a link.
So we could simply prompt the user if he wants us to display the license and if so present him with that URL or its contents.

**Comment by mstormi** [comment]
Created PR#459.

**Comment by mstormi** [comment]
@BClark09 mind to check [PR #459](https://github.com/openhab/openhabian/pull/459)

**Comment by BClark09** [comment]
Will do, but will only have time to properly check it in the weekend.

**Comment by EliasGabrielsson** [comment]
@mstormi Do you have a source on the deprecation of the deb repository from Azul?
Otherwise, I assume that the rep is still maintained and we can close this issue and related PR:s.

**Comment by mstormi** [comment]
No real source but the repo has not been updated in ages (6+ months) meanwhile they did create and offer new packages for download. I think it's reasonably safe to assume they don't maintain it any more and even if they did it would make sense to switch to package downloads as these are way more up to date and reliable.

**Comment by EliasGabrielsson** [comment]
I don't agree on that. 
6+ month is not a long time in an enterprise context so we shall not hesitated to draw quick conclusions. The documentation still have APT as an installation case so we following it.

It would also be a stupid not to use built in package manager in Debian/Ubuntu as new systems will be without updates after a while.

I am closing this for now. Have not anything happened for another 6 month I think it would be time to look for options.

**Comment by BClark09** [comment]
Hi @EliasGabrielsson, whilst I agree that using a Debian repository would be best, the apt repo hasnt been updated for too long.

Currently, openHABian installs a version of Java lower than [the minimum we recommend](https://www.openhab.org/docs/installation/#prerequisites) so in my opinion this is a change we must unfortunately make.

**Comment by ThomDietrich** [comment]
It's too bad they do not update their repository more frequent. Following your discussion I agree to switch over to tar-based installation.

Regarding the whether or not: Did any of you ask? 🙃 http://zulu.org/forum

**Comment by EliasGabrielsson** [comment]
@BClark09  I rest my case. If we are shipping Openhabian with *lower* version then the recommended / integrated version of java for OpenHAB a natural consequence is definitely to update the java package. Even if it mean to use a TAR installation.

I would prefer that the new script using similar mechanics as the node script to always fetching the latest version i.e not hardcoded versions. (See snippet of code below:)
```
f=$(wget -qO- https://nodejs.org/download/release/latest-dubnium/ | grep "armv6l.tar.gz" | cut -d '"' -f 2);
cond_redirect wget -O /tmp/nodejs-arm.tar.gz https://nodejs.org/download/release/latest-dubnium/$f 2>&1 || FAILED=1
```
What do you think about that?
Then the script can also be used to upgrade existing TAR based installation with the latest version of java when it's released.

**Comment by mstormi** [comment]
We all would like to use a link to the latest version instead of a hardcoded number but it doesn't exist to the best of my knowledge. If anyone discovers it let me know else we should proceed with hardcoding. Am 29.12.2018 00:53 schrieb Elias Gabrielsson <notifications@github.com>:@BClark09  I rest my case. If we are shipping Openhabian with lower version then the recommended / integrated version of java for OpenHAB a natural consequence is definitely to update the java package. Even if it mean to use a TAR installation.
I would prefer that the new script using similar mechanics as the node script to always fetching the latest version i.e not hardcoded versions. (See snippet of code below:)
f=$(wget -qO- https://nodejs.org/download/release/latest-dubnium/ | grep "armv6l.tar.gz" | cut -d '"' -f 2);
cond_redirect wget -O /tmp/nodejs-arm.tar.gz https://nodejs.org/download/release/latest-dubnium/$f 2>&1 || FAILED=1

What do you think about that?
Then the script can also be used to upgrade existing TAR based installation with the latest version of java when it's released.

**Comment by EliasGabrielsson** [comment]
Tried to figure out a "latest link" today with no success hardcoded version is the way to go. @mstormi

**Comment by GruensFroeschli** [comment]
While trying to install this on amd64 the installation fails.
There doesn't seem to be a zulu-embeeded version for amd64.

The path to the file is currently hardcoded to
http://cdn.azul.com/zulu-embedded/bin/${JAVA}.tar.gz

However for amd64 it should be
http://cdn.azul.com/zulu/bin/${JAVA}.tar.gz

The whole discussion above was that the repository is not up to date for zulu-embedded, but it seems to be up to date for the "normal" version.
Right now I've added the repository by hand and did a manual installation.
I guess it would make sense to keep the repository for amd64 installations and only do the .tar.gz installation for arm.

**Comment by mstormi** [comment]
Yes the link needs a fix but it's a plain wrong view of the world you reveal.
The "normal" version is the ARM one. _Man sollte nicht von sich auf andere schliessen._

**Comment by GruensFroeschli** [comment]
Seen from azul?
They way their page is set up i was under the impression that x86 is their main target and the embedded version is a side-product.

Seen from openhabian, arm is definitively the main target.

Regardless of that, my point is that I think for x86 the repository installation should stay.
The now merged PR #459 removed this option.
